src/parse/parser.rs

```diff
@@ -76,6 +64,19 @@ impl Input {
             Some((current, remaining))
         }
     }
+
+    /// Insert some items to the Input items just after current parsing cursor.
+    /// Usually used by replaced items recovering.
+    pub(crate) fn insert(&mut self, insert_items: &[&str]) {
+        self.items = {
+            let mut new_items: Vec<OsString> = insert_items.iter().map(OsString::from).collect();
+            for unparsed_item in &self.items[self.cursor..self.items.len()] {
+                new_items.push(unparsed_item.clone());
+            }
```

Comment:
```
            for i in self.cursor..self.items.len() {
                new_items.push(self.items[i].clone());
            }
```

Reply:
Not sure which is more readable but I felt this one is.

Reply:
Wait what I wanted to write is `for unparsed_item in &self.items[self.cursor..] {` would this be better? :)

Reply:
Yes